### ISSUE TITLE
vscode remote containers fails to open on Linux

### ISSUE THREAD

**Issue by reaper8055** [microsoft/vscode-remote-release#4263]
- VSCode Version:
Version: 1.52.1
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:32:10.090Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0

- OS Version:
OS: Arch Linux x64 5.4.86-1-lts

Steps to Reproduce:

(Not sure if this can be reproduced)

1. Install docker on linux and then install remote container extension for vscode 
2. Add a .devcontainer folder to the project folder
3. Open command pallet and search for "reopen in container"

This would fail with the following error:
https://gist.github.com/reaper8055/cfaf463b0a33f290a342c119923192bb

Things I have tried:
1. I tried with my devcontainer config that I used on my mac(which died 2 days back) and it used to work there flawlessly: https://github.com/reaper8055/devcontainer
2. Tried creating devcontainer.json from the available templates using `Ctrl + Shift + p` then `Add Development Container Configuration files...` but eneded up with same error which was:
```
[PID 174307] [876 ms] Start: Downloading VS Code Server
[2021-01-04T18:33:21.788Z] [PID 174307] [2589 ms] Error: connect ECONNREFUSED 0.0.0.0:443
	at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1134:16)
```
3. Tried uninstalling and reinstalling vscode and docker but still same issue. 

Does this issue occur when all extensions are disabled?: Yes/No

Yes the issue occurs even when I run `code --disable-extensions` but then I have to install remote container extension for this. 

Here are some additional information:
> docker version 
```
Client:
 Version:           20.10.1
 API version:       1.41
 Go version:        go1.15.6
 Git commit:        831ebeae96
 Built:             Tue Dec 15 22:25:01 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.1
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.15.6
  Git commit:       f0014860c1
  Built:            Tue Dec 15 22:24:28 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b.m
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
> docker-compose version
```
docker-compose version 1.27.4, build unknown
docker-py version: 4.4.1
CPython version: 3.9.1
OpenSSL version: OpenSSL 1.1.1i  8 Dec 2020
```

**Comment by chrmarti** [comment]
Are you using the community AUR package of VS Code?

**Comment by reaper8055** [comment]
I'm using the [AUR](https://aur.archlinux.org/packages/visual-studio-code-bin/?O=10&PP=10) for VS Code. The AUR package name is visual-studio-code-bin.

**Comment by chrmarti** [comment]
The community build is missing some of the required bits. These are currently only available with the official VS Code builds.

**Comment by reaper8055** [comment]
Ohh, okay... 🥺 Just so we're on the same page, I am using the VSCode shipped by microsoft which can access all the extension gallery, etc. I am not using the [open-source](https://archlinux.org/packages/community/x86_64/code/.) build of the VSCode which is build from the vscode git repo.

Official VS Code Builds -- meaning the ones available to download on this https://code.visualstudio.com/download?

**Comment by reaper8055** [comment]
So I tried the same on KDE Neon which is based on Ubuntu 20.04 LTS.

> How I installed VS Code:
1. sudo apt update && sudo apt upgrade -y
2. sudo apt install software-properties-common apt-transport-https wget
3. wget -q https://packages.microsoft.com/keys/microsoft.asc -O- | sudo apt-key add -
4. sudo add-apt-repository "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main"
5. sudo apt install code

$ vscode version:
```
Version: 1.52.1
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:32:10.090Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.0-59-generic
```
$ cat /etc/os-release
```
Version: 1.52.1
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:32:10.090Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.0-59-generic
```
> How I installed docker:
1. sudo apt install docker docker-compose -y 

$ docker version 
```
Client:
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.13.8
 Git commit:        afacb8b7f0
 Built:             Wed Oct 14 19:43:43 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       afacb8b7f0
  Built:            Wed Oct 14 16:41:21 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.3-0ubuntu2
  GitCommit:        
 runc:
  Version:          spec: 1.0.1-dev
  GitCommit:        
 docker-init:
  Version:          0.18.0
  GitCommit:        
```

$ docker-compose version
```
docker-compose version 1.25.0, build unknown
docker-py version: 4.1.0
CPython version: 3.8.5
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
```

> How I started devcontainers on VS Code
1. Opened VS Code
2. Installed remote containers from the extenstion gallery
3. Ctrl + Shift + P : Add Development Container Configuration files
4. Reopen in container
5. Same error message:
	```
	[2021-01-06T14:19:08.876Z] [PID 29675] [51698 ms] Start: Downloading VS Code Server
	[2021-01-06T14:19:10.365Z] [PID 29675] [53187 ms] Error: connect ECONNREFUSED 0.0.0.0:443
	at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1134:16)

	```

Complete error log:
https://gist.github.com/reaper8055/330bfdc8cd29cbe389a9cacc3c752a35

I'm still getting the same issue on ubuntu20.04. Is this supposed to happen?

**Comment by chrmarti** [comment]
That should work. Are you behind a proxy? It looks like `update.code.visualstudio.com` resolved to 0.0.0.0 by DNS. Not sure how that could happen or if something else is going on.

**Comment by reaper8055** [comment]
It looks like it was a DNS issue. I was using dnscrpt-proxy in my USG pro and it was resolving that and even the vscode download URLs from Microsoft's website to 0.0.0.0. Don't know why. I switched back to dnsmasq, flushed the cache and it started working. 

Thanks a bunch for pointing the dns thing out. And apologies for creating issue without proper diagnosis on my part 😔.
